kernel/src/scan/mod.rs

```diff
@@ -405,20 +448,28 @@ pub fn transform_to_logical(
     let (all_fields, _read_fields, have_partition_cols) = get_state_info(
         &global_state.logical_schema,
         &global_state.partition_columns,
+        &global_state.column_mapping_mode,
     );
     let read_schema = Arc::new(global_state.read_schema.clone());
-    if have_partition_cols {
-        // need to add back partition cols
+    if have_partition_cols || global_state.column_mapping_mode != ColumnMappingMode::None {
+        // need to add back partition cols and/or fix-up mapped columns
```

Comment:
Yep, we could do that, at the cost of slowing down queries which don't have partition cols _or_ mapped cols. This is there because in the case no fixup is needed, you can just return `Ok(data)` below, and do zero work.

Everything would "just work" if we removed this `if` and just always eval'd the expression, it's just an optimization, but one that feels like it's worth doing.

Reply:
Update: Never mind... We need that `Expression:column` if column mapping is enabled.